### PR TITLE
[Remove Stream/Optional usage] LPS-178143 portal-workflow-kaleo-service

### PR DESCRIPTION
**Pull request by JanisZhang**
Resend. Thanks!

previous: https://github.com/liferay-core-infra/liferay-portal/pull/3839
https://issues.liferay.com/browse/LPS-178143

### CODE REVIEW

**Comment by JanisZhang** [comment]
ci:test:workflow

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v3_1_1/KaleoNotificationUpgradeProcess.java [hunk 1]
```diff
@@ -48,16 +48,19 @@ protected void doUpgrade() throws Exception {
...
-						notificationType, "private-message")
-				).toArray(
-					String[]::new
-				);
+				String[] notificationTypes = TransformUtil.transform(
+					StringUtil.split(resultSet.getString("notificationTypes")),
+					notificationType -> {
+						if (Objects.equals("im", notificationType) ||
+							Objects.equals(
+								notificationType, "private-message")) {
```
Comment (dantewang) [inline modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v3_1_1/KaleoNotificationUpgradeProcess.java:56]: @ling-alan-huang Here we try to put the looped item `notificationType` the second operand in the equals comparison, but SF automatically changes the one in line 55~56 back. Could you take a look why it's not enforcing the first one?

**Comment by dantewang** [comment]
Hold on for now due to the auto SF inconsistency.

**Review by dantewang — APPROVED** [review]

File: modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v3_1_1/KaleoNotificationUpgradeProcess.java (hunk 1 above)
Comment (ling-alan-huang) [reply modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v3_1_1/KaleoNotificationUpgradeProcess.java:56]: I will have a look at it tomorrow.

Comment (ling-alan-huang) [reply modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v3_1_1/KaleoNotificationUpgradeProcess.java:56]: @qz1010 Can you have a look at it? I am off today.

**Comment by dantewang** [comment]
ci:forward

cc @shuyangzhou @tinatian

**Comment by ling-alan-huang** [comment]
SF fix: https://github.com/ling-alan-huang/liferay-portal/pull/1737

**Comment by JanisZhang** [comment]
> SF fix: https://github.com/ling-alan-huang/liferay-portal/pull/1737

Thanks @ling-alan-huang !

**Pull request closed without merging**